File: src/Cxx/IO/JPEGWriter.cxx
#include <vtkSmartPointer.h>
#include <vtkImageCanvasSource2D.h>
#include <vtkJPEGWriter.h>

int main(int vtkNotUsed(argc), char* vtkNotUsed(argv)[])
{
  std::string outputFilename = "output.jpg";

  // Create a 100x100 image to save into the jpeg file
  int extent[6] = { 0, 99, 0, 99, 0, 0 };
  vtkSmartPointer<vtkImageCanvasSource2D> imageSource =
    vtkSmartPointer<vtkImageCanvasSource2D>::New();
  imageSource->SetExtent( extent );
  imageSource->SetScalarTypeToUnsignedChar(); // vtkJPEGWriter only accepts unsigned char input
  imageSource->SetNumberOfScalarComponents( 3 ); // 3 color channels: Red, Green and Blue

  // Fill the whole image with a blue background
  imageSource->SetDrawColor( 0, 127, 255 );
  imageSource->FillBox( extent[0], extent[1], extent[2], extent[3] );

  // Paint a 30x30 white square into the image
  imageSource->SetDrawColor( 255, 255, 255 );
  imageSource->FillBox( 40, 70, 20, 50 );

  vtkSmartPointer<vtkJPEGWriter> writer =
    vtkSmartPointer<vtkJPEGWriter>::New();
  writer->SetFileName( outputFilename.c_str() );
  writer->SetInputConnection( imageSource->GetOutputPort() );
  writer->Write();

  return EXIT_SUCCESS;
}

File: src/Java/IO/JPEGWriter.java
import vtk.vtkImageCanvasSource2D;
import vtk.vtkJPEGWriter;
import vtk.vtkNamedColors;
import vtk.vtkNativeLibrary;

public class JPEGWriter 
{

  //-----------------------------------------------------------------
  // Load VTK library and print which library was not properly loaded
  static 
  {
    if (!vtkNativeLibrary.LoadAllNativeLibraries()) 
    {
      for (vtkNativeLibrary lib : vtkNativeLibrary.values()) 
      {
        if (!lib.IsLoaded()) 
        {
          System.out.println(lib.GetLibraryName() + " not loaded");
        }
      }
    }
    vtkNativeLibrary.DisableOutputWindow(null);
  }
  // -----------------------------------------------------------------
  
  public static void main(String args[]) 
  {
	  vtkNamedColors colors = new vtkNamedColors();
	  
	  double bkgColor[] = new double[] {255, 0,0,1};
	  double boxColor[] = new double[] {255,255,0,1};
	  
	  colors.SetColor("bkgColor", bkgColor);
	  colors.SetColor("boxColor", boxColor);
	  
	  
	// Provide default values.
	String FileName = "Output.jpg";
	for(int i = 0; i < args.length; ++i)
	{
	      switch (i) 
	      {
	        case 0:
	        		FileName = args[i];
	        		break;
	      }
	}

    vtkImageCanvasSource2D ImageSource = new vtkImageCanvasSource2D();
    ImageSource.SetExtent(0, 99, 0, 99, 0, 0);
    
    //vtkJPEGWriter only accepts unsigned char input
    ImageSource.SetScalarTypeToUnsignedChar();
    
    //3 color channels: Red, Green and Blue
    ImageSource.SetNumberOfScalarComponents(3);

    //Fill the whole image with a blue background
    ImageSource.SetDrawColor(bkgColor);
    ImageSource.FillBox(0, 99, 0, 99);

    //Paint a 30x30 white square into the image
    ImageSource.SetDrawColor(boxColor);
    ImageSource.FillBox(40, 70, 20, 50);

    vtkJPEGWriter Writer = new vtkJPEGWriter();
    Writer.SetFileName(FileName);
    Writer.SetInputConnection(ImageSource.GetOutputPort());
    Writer.Write();

  }
}
